File: wp8/Google Balls/Vector.cs
namespace Google_Balls
{
    public class Vector
    {
        public double X { get; set; }
        public double Y { get; set; }
        public double Z { get; set; }

        public Vector(double x = 0, double y = 0, double z = 0)
        {
            X = x;
            Y = y;
            Z = z;
        }

        public void AddX(double x)
        {
            X += x;
        }

        public void AddY(double y)
        {
            Y += y;
        }

        public void AddZ(double z)
        {
            Z += z;
        }

        public void Set(double x, double y, double z)
        {
            X = x;
            Y = y;
            Z = z;
        }
    }
}

File: android/app/src/main/java/com/weenie/googleballs/Vector.java
package com.weenie.googleballs;

public class Vector {
    public float x;
    public float y;
    public float z;

    public Vector(float x, float y) {
        this.x = x;
        this.y = y;
        this.z = 0;
    }

    public Vector(float x, float y, float z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    public void addX(float x) {
        this.x += x;
    }

    public void addY(float y) {
        this.y += y;
    }

    public void addZ(float z) {
        this.z += z;
    }

    public void set(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public void set(float x, float y, float z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
}
